### PR TITLE
For #7075: Expand toolbar on navigation or load request

### LINKED ISSUE #7075: show toolbar with loading progress when toolbar is hidden and url is tapped
## Steps to reproduce

1. Search any term on google (ex. search android).
2. scroll up so that the toolbar is hidden.
3. Tap on any search result.
4. check that the toolbar is not revealed after URL tap to show the loading progress.

### Expected behavior

1. The toolbar should reveal itself when URL is tapped.
2. Toolbar is revealed on url tap in Firefox Android App (Fenix).
3. This behaviour is especially important in poor network connection, when it gets unclear for user if the URL is tapped properly or not.

### Actual behavior

1. when user taps on link to navigate to that URL and the toolbar is hidden bacause of scrolling the toolbar doesn't become visible to show progress.

### Device information

* Android device: 12
* Focus version: 100.3.0

### PR DESCRIPTION
**Pull request by ryg-git**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

May fix #7075

issue-7075

https://user-images.githubusercontent.com/69393483/169834083-ca0897e0-d59e-46c3-ac39-93da284b299c.mp4

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt [hunk 1]
```diff
@@ -275,6 +277,23 @@ class BrowserToolbarIntegration(
         }
     }
 
+    @VisibleForTesting
```
Comment (Mugurell) [inline app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt:280]: This annotation allows relaxing the visibility specifically for tests.
Will you add also add the tests for this method or can the annotation be removed and the method be marked as private?

Comment (ryg-git) [reply app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt:280]: ok I will remove the annotation

**Review by Mugurell — COMMENTED** [review]
LGTM. Thank you!
Great that you also included the small video to showcase the result!

Can you please squash your commits to just one and update the commit message and PR title to include a reference to the ticket this patch intends to fix, making it easier to find it later?
As described here - https://github.com/mozilla-mobile/shared-docs/blob/main/android/CONTRIBUTING_code.md#creating-a-pull-request The resulting commit title would have the form
> For #7075: Expand toolbar on navigation or load request

**Review by Mugurell — APPROVED** [review]
LGTM! Thank you!

**Comment by Mugurell** [comment]
@ryg-git There were some formatting issues causing the linting tasks to fail which I've addressed and will land your PR in https://github.com/mozilla-mobile/focus-android/pull/7111.
It's a bit of a more cumbersome process but you are still the author of the commit, thank you for that!

**Pull request closed without merging**